shapely/tests/test_io.py

```diff
@@ -511,6 +618,45 @@ def test_to_wkb_z():
     assert shapely.to_wkb(point, byte_order=1) == expected_wkb_z
     assert shapely.to_wkb(point, output_dimension=2, byte_order=1) == expected_wkb
     assert shapely.to_wkb(point, output_dimension=3, byte_order=1) == expected_wkb_z
+    if shapely.geos_version >= (3, 12, 0):
+        assert shapely.to_wkb(point, output_dimension=4, byte_order=1) == expected_wkb_z
+
+
+def test_to_wkb_m():
+    # POINT M (1 2 4)
+    point = shapely.from_wkb(struct.pack("<BI3d", 1, 1 | EWKBM, 1.0, 2.0, 4.0))
+
+    expected_wkb = struct.pack("<BI2d", 1, 1, 1.0, 2.0)
+    expected_wkb_m = struct.pack("<BI3d", 1, 1 | EWKBM, 1.0, 2.0, 4.0)
+    if shapely.geos_version < (3, 12, 0):
+        # previous behavior was to ignore M, treat as 2D
+        expected_wkb_m = expected_wkb
+
+    assert shapely.to_wkb(point, byte_order=1) == expected_wkb_m
+    assert shapely.to_wkb(point, output_dimension=2, byte_order=1) == expected_wkb
+    assert shapely.to_wkb(point, output_dimension=3, byte_order=1) == expected_wkb_m
```

Comment:
So in this case (of M and not ZM), "M" is regarded as the third dimension? ( I would have naively expected this to drop the M values)

Reply:
The first and second dims are always X and Y, but beyond that there are no guarantees. This is why I'm advocating renaming away from "3D" as it's ambiguous.